### PR TITLE
Add a page about common methods in Godot's C++ codebase

### PR DESCRIPTION
**Pull request by Calinou**
This is geared towards beginner engine contributors, but it could also be useful to those writing custom C++ modules.

### CODE REVIEW

File: development/cpp/common_engine_methods_and_macros.rst [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+    message argument.
+
+    Also, always try to return processable data so the engine can keep running
+    well.
+
+.. code-block:: cpp
+
+    // Conditionally prints an error message and returns from the function.
+    // Use this in methods which don't return a value.
+    ERR_FAIL_COND_MSG(vformat("Couldn't load mesh at: %s", path));
```
Comment (bojidar-bg) [inline development/cpp/common_engine_methods_and_macros.rst:202]: `ERR_FAIL_COND_MSG` takes two parameters; `ERR_FAIL_COND_V_MSG` (below) takes 3.

File: development/cpp/common_engine_methods_and_macros.rst [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+
+    // Conditionally prints an error message and returns `0` from the function.
+    // Use this in methods which must return a value.
+    ERR_FAIL_COND_V_MSG(0, "Couldn't calculate the rectangle's area.");
+
+    // Prints an error message if ``index`` is less than 0 or greater than
```
Comment (bojidar-bg) [inline development/cpp/common_engine_methods_and_macros.rst:208]: The condition is `< 0 || >= max`, so should technically be "less than 0 or greater or equal to max".

**Comment by clayjohn** [comment]
Great work! This will be super helpful for new contributors.

**Comment by mhilbrunner** [comment]
Thank you, this would've been useful back when I started :)

**Pull request merged**